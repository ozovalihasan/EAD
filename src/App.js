import React, { useState } from 'react';
import styled from 'styled-components';
import { DragDropContext } from 'react-beautiful-dnd';
import initialData from './initial-data';
import Column from './column';

const App = () => {
  // setTimeout(() => { console.warn('hasan'); }, 0);
  const [data, setData] = useState(initialData);

  // const onDragStart = () => {
  //   document.body.style.color = 'orange';
  //   document.body.style.transition = 'background-color 0.2s ease';
  // };

  // const onDragUpdate = (update) => {
  //   const { destination } = update;
  //   const opacity = destination
  //     ? destination.index / Object.keys(data.tasks).length
  //     : 0;
  //   document.body.style.backgroundColor = `rgba(153, 141, 217, ${opacity})`;
  // };
  const onDragStart = (start) => {
    const homeIndex = data.columnOrder.indexOf(start.source.droppableId);

    setData({
      ...data,
      homeIndex,
    });
  };
  const onDragEnd = (result) => {
    const {
      destination, source, draggableId,
    } = result;

    setData({
      ...data,
      homeIndex: null,
    });
    if (!destination) {
      return;
    }

    if (
      destination.droppableId === source.droppableId
      && destination.index === source.index
    ) {
      return;
    }

    const home = data.columns[source.droppableId];
    const foreign = data.columns[destination.droppableId];

    if (home === foreign) {
      const newTaskIds = Array.from(home.taskIds);
      newTaskIds.splice(source.index, 1);
      newTaskIds.splice(destination.index, 0, draggableId);

      const newHome = {
        ...home,
        taskIds: newTaskIds,
      };

      const newData = {
        ...data,
        columns: {
          ...data.columns,
          [newHome.id]: newHome,
        },
      };

      setData(newData);
      return;
    }

    const homeTaskIds = Array.from(home.taskIds);
    homeTaskIds.splice(source.index, 1);
    const newHome = {
      ...home,
      taskIds: homeTaskIds,
    };

    const foreignTaskIds = Array.from(foreign.taskIds);
    foreignTaskIds.splice(destination.index, 0, draggableId);
    const newForeign = {
      ...foreign,
      taskIds: foreignTaskIds,
    };

    const newData = {
      ...data,
      columns: {
        ...data.columns,
        [newHome.id]: newHome,
        [newForeign.id]: newForeign,
      },
    };
    setData(newData);
  };

  return (
    <div className="App">
      <DragDropContext
        onDragStart={onDragStart}
        onDragEnd={onDragEnd}
      >
        <Container>

          {data.columnOrder.map((columnId, index) => {
            const column = data.columns[columnId];
            const tasks = column.taskIds.map(
              (taskId) => data.tasks[taskId],
            );
            const isDropDisabled = index < data.homeIndex;

            return (
              <Column
                key={column.id}
                column={column}
                tasks={tasks}
                isDropDisabled={isDropDisabled}
              />
            );
          })}

        </Container>

      </DragDropContext>

    </div>
  );
};

const Container = styled.div`
  display: flex;
`;

export default App;
