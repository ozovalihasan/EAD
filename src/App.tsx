import React, { useState, useRef, useCallback } from 'react';
import ReactFlow, {
  ReactFlowProvider,
  Controls,
  Node,
  ReactFlowInstance,
} from 'react-flow-renderer';

import Sidebar from './Sidebar';
import './index.css';
import EntityNode, { EntityNodeDataType } from './EntityNode';
import useStore from './zustand/store';
import HasManyEdge from './HasManyEdge';
import HasOneEdge from './HasOneEdge';
import ThroughEdge from './ThroughEdge';
import ConnectionLine from './ConnectionLine';



const nodeTypes = {
  entity: EntityNode,
};
const edgeTypes = {
  hasMany: HasManyEdge,
  hasOne: HasOneEdge,
  through: ThroughEdge,
};

const App = () => {
  
  const { 
    nodes, 
    edges, 
    onNodesChange, 
    onEdgesChange, 
    onConnect, 
    addNode, 
    onConnectStart, 
    onConnectEnd, 
    onNodeMouseEnter, 
    onNodeMouseLeave,
    onEdgeMouseEnter, 
    onEdgeMouseLeave,
  } = useStore();
  const reactFlowWrapper = useRef<HTMLDivElement>(null);
  const [reactFlowInstance, setReactFlowInstance] = useState<ReactFlowInstance<any, any> | null>(null);


  const onDragOver = useCallback((event: React.DragEvent<HTMLDivElement>) => {
    event.preventDefault();
    event.dataTransfer.dropEffect = 'move';
  }, []);

  const onDrop = useCallback(
    (event: React.DragEvent<HTMLDivElement>) => {
      event.preventDefault();

      let reactFlowBounds: DOMRect = (reactFlowWrapper.current as HTMLDivElement).getBoundingClientRect();
      
      const type = event.dataTransfer.getData('application/reactflow');
      const tableId = event.dataTransfer.getData('tableId');

      // check if the dropped element is valid
      if (typeof type === 'undefined' || !type) {
        return;
      }

      let position: {x: number, y: number} = (reactFlowInstance as ReactFlowInstance).project({
          x: event.clientX - reactFlowBounds.left,
          y: event.clientY - reactFlowBounds.top,
        });

      
      let name = useStore.getState().tables[tableId].name;

      const newNode: Node<EntityNodeDataType> = {
        id: useStore.getState().idCounter.toString(),
        type: "entity",
        position,
        data: { tableId, name },
      };
      

      addNode(newNode);
    },
    [reactFlowInstance]
  );


  return (
    <div  className="h-screen w-screen bg-first-100 flex flex-row flex-grow">
      <ReactFlowProvider>
        <div className="h-full flex-grow" ref={reactFlowWrapper}>

          <ReactFlow
            nodes={nodes}
            edges={edges}
            onNodesChange={onNodesChange}
            onEdgesChange={onEdgesChange}
            onConnect={onConnect}
            onInit={setReactFlowInstance}
            onDrop={onDrop}
            onConnectStart={onConnectStart}
            onConnectEnd={onConnectEnd}
            onNodeMouseEnter={onNodeMouseEnter}
            onNodeMouseLeave={onNodeMouseLeave}
            onEdgeMouseEnter={onEdgeMouseEnter}
            onEdgeMouseLeave={onEdgeMouseLeave}
            onDragOver={onDragOver}
            edgeTypes={edgeTypes}
            deleteKeyCode={"Delete"}
            connectionLineComponent={ConnectionLine }
            elevateEdgesOnSelect={true}
            fitView
            nodeTypes={nodeTypes}
            snapToGrid={true}
            
          >
            <Controls />

          </ReactFlow>
        </div>
        <Sidebar />
      </ReactFlowProvider>
    </div>

  )
}

export default App
