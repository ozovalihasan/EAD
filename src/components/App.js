import React from 'react';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import { useDispatch, useSelector } from 'react-redux';
import { library } from '@fortawesome/fontawesome-svg-core';
import {
  faArrowsAlt, faExpandAlt, faCompressAlt, faEllipsisH, faEllipsisV, faFlag, faClone,
  faPlus, faPlaneDeparture, faPlaneArrival, faTimes,
} from '@fortawesome/free-solid-svg-icons';
import { faGithub } from '@fortawesome/free-brands-svg-icons';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Model from './Model';
import {
  removeItem,
  addItem,
  moveItem,
  updateDraggedItemId,
  resetState,
  installState,
  idCountIncrease,
  toggleExpandAll,
  toggleCompactMode,
  updateRestrictedDropId,
} from '../redux';
import saveJSON from './saveJSON';
import colors from './colors';

library.add(faArrowsAlt, faExpandAlt, faCompressAlt, faEllipsisH,
  faEllipsisV, faFlag, faClone, faGithub, faPlus, faPlaneDeparture, faPlaneArrival, faTimes);

const App = () => {
  const {
    items, dragDropCategory, idCount, expandAll, compactMode,
  } = useSelector((state) => state.block);

  const startingId = compactMode ? 9 : 0;
  const dispatch = useDispatch();

  const checkDragDropCategory = (dragId, dropId) => (
    dragDropCategory[items[dragId].category].includes(items[dropId].category)
  );

  const stateBlock = useSelector((state) => state.block);

  const saveBlocks = () => {
    localStorage.block = JSON.stringify(stateBlock);
  };

  const onDragStart = (result) => {
    dispatch(updateDraggedItemId(result.draggableId));
  };

  const onDragEnd = (result) => {
    dispatch(updateDraggedItemId(-1));
    const {
      destination, draggableId, source,
    } = result;

    if (!destination) {
      if (!(items[draggableId].factory)) {
        dispatch(removeItem(source.droppableId, source.index, draggableId));
      }
      return;
    }

    if (!checkDragDropCategory(draggableId, destination.droppableId)) {
      return;
    }

    if (items[draggableId].factory) {
      dispatch(addItem(draggableId, destination.droppableId, destination.index, idCount));
      if (items[draggableId].association) {
        dispatch(updateRestrictedDropId(idCount, null));
      }
      dispatch(idCountIncrease());
      return;
    }

    if (items[destination.droppableId].entityContainer
      && items[destination.droppableId].subItemIds.length > 0
      && (items[items[destination.droppableId].subItemIds[0]].category
        !== items[draggableId].category
      )) {
      return;
    }
    dispatch(moveItem(
      draggableId,
      destination.droppableId,
      destination.index,
      source.droppableId,
      source.index,
    ));
  };

  return (
    <MainContainer className="App">
      <LogoButtons>
        <Logo src={`${process.env.PUBLIC_URL}/images/ead-logo.svg`} alt="EAD logo" />
        <Button
          onClick={() => saveJSON(items, 'EAD.json')}
          type="button"
        >
          Download EAD
        </Button>

        <CompactMode
          onClick={() => dispatch(toggleCompactMode())}
          type="button"
          compactMode={compactMode}
        >
          Compact Mode
        </CompactMode>

        {compactMode
        || (
          <>
            <Button
              onClick={() => dispatch(resetState())}
              type="button"
            >
              Reset
            </Button>
            <Button
              onClick={() => saveBlocks()}
              type="button"
            >
              Save
            </Button>
            <Button
              onClick={() => dispatch(installState())}
              type="button"
            >
              Install Saved Data
            </Button>

            <ExpandAllButton
              onClick={() => dispatch(toggleExpandAll())}
              type="button"
              expandAll={expandAll}
            >
              Expand All
            </ExpandAllButton>

            <GithubLink href="https://github.com/ozovalihasan/ead">
              <FontAwesomeIcon icon={faGithub} size="2x" />
            </GithubLink>
          </>
        )}
      </LogoButtons>
      <DragDropContext
        onDragStart={onDragStart}
        onDragEnd={onDragEnd}
      >
        <Droppable
          droppableId="OuterDrop"
          direction="vertical"
          isDropDisabled
        >

          { (provided) => (
            <div
              {...provided.droppableProps}
              ref={provided.innerRef}
            >
              <Model
                parentId={startingId}
                item={items[startingId]}
                allItems={items}
                index={startingId}
                checkDragDropCategory={checkDragDropCategory}
              />
              {provided.placeholder}
            </div>
          )}

        </Droppable>
      </DragDropContext>
    </MainContainer>
  );
};

const MainContainer = styled.div`
  font-family: Arial, Helvetica, sans-serif;
`;

const LogoButtons = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
`;

const Logo = styled.img`
  width: 60px;
  padding-right: 40px;
`;

const Button = styled.button`
  border-radius: 5px;
  border: 1px solid ${colors.disabled};
  background-color: transparent;
  margin: 10px;
  padding: 10px;
  transition: background-color 0.2s;

  &:hover {
    cursor: pointer;
    background-color: ${colors.entity};
  }
`;

const ExpandAllButton = styled(Button)`
  background-color: ${(props) => (props.expandAll && colors.association)};
`;

const CompactMode = styled(Button)`
  background-color: ${(props) => (props.compactMode && colors.entity)};
`;

const GithubLink = styled.a`
  margin: 0 10px;
  color: #000;

  &:hover {
    color: ${colors.chosen};
  }
`;
export default App;
