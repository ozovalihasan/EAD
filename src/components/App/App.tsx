import {
  Sidebar, 
  Navbar,
  SidebarHandle,
  MainReactFlow
} from "@/components"
import { useCustomizationStore } from '@/zustandStore/customizationStore';

export const App = () => {
  
  const locationSidebar = useCustomizationStore((state) => state.locationSidebar)
  
  return (

    <div className="text-zinc-600 dark:text-zinc-100 font-default bg-zinc-50 dark:bg-zinc-700 flex flex-col h-screen">
      <Navbar />
      <div  className={`h-[calc(100vh-5rem)] w-screen  flex flex-grow ${locationSidebar === "left" ? "flex-row" : "flex-row-reverse"}`}>
        <Sidebar />
        <SidebarHandle />
        <MainReactFlow />
      </div>
    </div>

  )
}