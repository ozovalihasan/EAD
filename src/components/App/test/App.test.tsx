import {  App } from '../App';
import { render, screen } from "@testing-library/react";
import useCustomizationStore from '@/zustandStore/customizationStore';

jest.mock('react-flow-renderer',  () => ({
  Sidebar:(
    () => (
      <>
        MockEADLogo
      </>
    )
  ),
}))

jest.mock('@/components',  () => ({
  Sidebar: (
    () => (
      <>
        MockSidebar
      </>
    )
  ),
  Navbar: (
    () => (
      <>
        MockNavbar
      </>
    )
  ),
  SidebarHandle: (
    () => (
      <>
        MockSidebarHandle
      </>
    )
  ),
  MainReactFlow: (
    () => (
      <>
        MockMainReactFlow
      </>
    )
  ),

}))
  

let renderReadyComponent: JSX.Element;

beforeEach(() => {
  renderReadyComponent = (
    <App />
  );
});

describe('<App />', () => {

  describe('if the sidebar is at the left of the screen', () => {

    beforeEach(() => {
      useCustomizationStore.setState({ 
        locationSidebar: "left"
      })
    })
    
    it('has a "flex-row" class name ', () => {
      render(renderReadyComponent );
  
      expect(screen.getByText(/MockSidebar/i).classList).toContain("flex-row");
    });

    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
  
  })

  describe('if the sidebar is at the right of the screen', () => {

    beforeEach(() => {
      useCustomizationStore.setState({ 
        locationSidebar: "right"
      })
    })

    it('has "flex-row-reverse" class name if the sidebar is at the right of the screen', () => {
      
      render(renderReadyComponent );
      expect(screen.getByText(/MockSidebar/i).classList).toContain("flex-row-reverse");
    });
    

    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
  
  })
  
  it('renders different components', () => {

    render(renderReadyComponent );

    expect(screen.getByText(/MockSidebar/i)).toBeInTheDocument();
    expect(screen.getByText(/MockNavbar/i)).toBeInTheDocument();
    expect(screen.getByText(/MockSidebarHandle/i)).toBeInTheDocument();
    expect(screen.getByText(/MockMainReactFlow/i)).toBeInTheDocument();
  });

  
  it('renders correctly', () => {
    const renderedContainer = render(renderReadyComponent );
    expect(renderedContainer).toMatchSnapshot();
  });

});