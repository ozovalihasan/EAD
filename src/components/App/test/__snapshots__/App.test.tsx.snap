// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> if the sidebar is at the left of the screen renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-zinc-600 dark:text-zinc-100 font-default bg-zinc-50 dark:bg-zinc-700 flex flex-col h-screen"
      >
        MockNavbar
        <div
          class="h-[calc(100vh-5rem)] w-screen  flex flex-grow flex-row"
        >
          MockSidebar
          MockSidebarHandle
          MockMainReactFlow
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-zinc-600 dark:text-zinc-100 font-default bg-zinc-50 dark:bg-zinc-700 flex flex-col h-screen"
    >
      MockNavbar
      <div
        class="h-[calc(100vh-5rem)] w-screen  flex flex-grow flex-row"
      >
        MockSidebar
        MockSidebarHandle
        MockMainReactFlow
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<App /> if the sidebar is at the right of the screen renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-zinc-600 dark:text-zinc-100 font-default bg-zinc-50 dark:bg-zinc-700 flex flex-col h-screen"
      >
        MockNavbar
        <div
          class="h-[calc(100vh-5rem)] w-screen  flex flex-grow flex-row-reverse"
        >
          MockSidebar
          MockSidebarHandle
          MockMainReactFlow
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-zinc-600 dark:text-zinc-100 font-default bg-zinc-50 dark:bg-zinc-700 flex flex-col h-screen"
    >
      MockNavbar
      <div
        class="h-[calc(100vh-5rem)] w-screen  flex flex-grow flex-row-reverse"
      >
        MockSidebar
        MockSidebarHandle
        MockMainReactFlow
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<App /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-zinc-600 dark:text-zinc-100 font-default bg-zinc-50 dark:bg-zinc-700 flex flex-col h-screen"
      >
        MockNavbar
        <div
          class="h-[calc(100vh-5rem)] w-screen  flex flex-grow flex-row"
        >
          MockSidebar
          MockSidebarHandle
          MockMainReactFlow
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-zinc-600 dark:text-zinc-100 font-default bg-zinc-50 dark:bg-zinc-700 flex flex-col h-screen"
    >
      MockNavbar
      <div
        class="h-[calc(100vh-5rem)] w-screen  flex flex-grow flex-row"
      >
        MockSidebar
        MockSidebarHandle
        MockMainReactFlow
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
